api/src/Security/Authenticator/HttpBearerAuthenticator.php

```diff
@@ -0,0 +1,159 @@
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Token\PostAuthenticationToken;
+use Symfony\Component\Security\Http\EntryPoint\AuthenticationEntryPointInterface;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * @final
+ */
+class HttpBearerAuthenticator implements AuthenticatorInterface, AuthenticationEntryPointInterface
+{
+    /**
+     * @var array<string, class-string<Signer>>
+     */
+    public const SIGN_ALGORITHMS = [
+        'HS256' => Signer\Hmac\Sha256::class,
+        'HS384' => Signer\Hmac\Sha384::class,
+        'HS512' => Signer\Hmac\Sha512::class,
+        'ES256' => Signer\Ecdsa\Sha256::class,
+        'ES384' => Signer\Ecdsa\Sha384::class,
+        'ES512' => Signer\Ecdsa\Sha512::class,
+        'RS256' => Signer\Rsa\Sha256::class,
+        'RS384' => Signer\Rsa\Sha384::class,
+        'RS512' => Signer\Rsa\Sha512::class,
+    ];
+
+    private UserProviderInterface $userProvider;
+    private string $realmName;
+    private string $payloadKey;
+    private ?string $key;
+    private Signer $signer;
+    private HttpClientInterface $client;
+    private LoggerInterface $logger;
+
+    public function __construct(UserProviderInterface $userProvider, string $realmName, string $payloadKey, string $algorithm, string $publicKey, HttpClientInterface $client = null, LoggerInterface $logger = null)
+    {
+        $this->userProvider = $userProvider;
+        $this->realmName = $realmName;
+        $this->payloadKey = $payloadKey;
+        $signerClass = self::SIGN_ALGORITHMS[$algorithm];
+        $this->signer = new $signerClass();
+        $this->key = $publicKey;
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null): Response
+    {
+        $response = new Response();
+        $response->headers->set('WWW-Authenticate', sprintf('Bearer realm="%s"', $this->realmName));
+        $response->setStatusCode(401);
+
+        return $response;
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $request->headers->has('AUTHORIZATION');
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        if (!class_exists(Configuration::class)) {
+            throw new RuntimeException('HttpBearerAuthenticator requires lcobucci/jwt, please run "composer require lcobucci/jwt" to install it.');
```

Comment:
could we introduce https://web-token.spomky-labs.com/ instead of lcobucci?

Reply:
This must be discussed with @chalasr and the Symfony team first. Symfony is currently using Lcobucci JWT (in the Mercure component). Let's avoid using two different dependencies doing the same thing if not necessary.

Reply:
I'm still not sure that having a jwt library integration in core is something we want. But if we do, adoption/popularity will probably be the most important criteria to make a choice, combined to quality. 
Sadly, web-token components are not as popular as they deserve yet so while lcobucci's lib suffer from the same bus factor, it has more dependents. As such, I think lcobucci/jwt keeps being the best fit.